lib/TPP/Dialect/Tpp/TppUtils.cpp

```diff
@@ -138,106 +135,105 @@ bool hasCopySemantics(linalg::LinalgOp linalgOp) {
   return hasOnlyOp<linalg::YieldOp>(linalgOp->getRegion(0));
 }
 
-// Returns the closest earlier user of a given operation op relative
-// to an another op user currentUser.
-// If there are no earlier users or the specified currentUser is invalid
-// e.g., it does not belong the def-use chain of op, return nullptr instead.
-static Operation *getPrevUser(Operation *op, Operation *currentUser) {
-  if (!op || !currentUser)
-    return nullptr;
-
-  // Iterate over the op users to find the given current user.
-  auto userIt = op->user_begin();
-  while (userIt != op->user_end()) {
-    if (*userIt == currentUser) {
-      // Given that the user_iterator visits the op users from the last to
-      // the first user, simply return the next user after the current user
-      // i.e. an earlier/previous operation.
-      const auto nextUser = ++userIt;
-      return nextUser != op->user_end() ? *nextUser : nullptr;
-    }
+// Returns true if the value is a constant float or integer.
+bool isValConstZero(Value val) {
+  return matchPattern(val, m_AnyZeroFloat()) || matchPattern(val, m_Zero());
+}
+
+// Prototypes
+static bool isZeroOp(Operation *, Operation *);
+
+// Returns true if the value represents a zero filled tensor.
+// Recurse into isZeroOp for defining ops if not imediately obvious
+// Looks past linalg generic's argument (which don't have defining ops)
+static bool isZeroValue(Value val, Operation *userOp) {
```

Comment:
Is the `userOp` needed for anything?

Reply:
Nope, it was left over from previous attempt at improving the "previous user" chain. Removed.